src/games/supported/Kaboom.cpp

```diff
@@ -0,0 +1,98 @@
+/* *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ */
+#include "Kaboom.hpp"
+
+#include "../RomUtils.hpp"
+#include <iostream>
+
+KaboomSettings::KaboomSettings() {
+    reset();
+}
+
+
+/* create a new instance of the rom */
+RomSettings* KaboomSettings::clone() const {
+    
+    RomSettings* rval = new KaboomSettings();
+    *rval = *this;
+    return rval;
+}
+
+
+/* process the latest information from ALE */
+void KaboomSettings::step(const System& system) {
+
+    // update the reward
+    reward_t score = getDecimalScore(0xA5, 0xA4, 0xA3, &system);
+    m_reward = score - m_score;
+    m_score = score;
+
+    // update terminal status
+    int lives = readRam(&system, 0xA1);
+    m_terminal = lives == 0x0 || m_score > 999991;
+}
+
+
+/* is end of game */
+bool KaboomSettings::isTerminal() const {
+
+    return m_terminal;
+};
+
+
+/* get the most recently observed reward */
+reward_t KaboomSettings::getReward() const {
+     return m_reward;
+}
+
+
+/* is an action part of the minimal set? */
+bool KaboomSettings::isMinimal(const Action &a) const {
+
+    switch (a) {
+        case PLAYER_A_NOOP:
+        case PLAYER_A_RIGHT:
+        case PLAYER_A_LEFT:
```

Comment:
The fire button is not needed to play the game, only to start it. I see it as outside of the scope of an agent that's trying to play the game (it has no effect on game state), but if you want me to add it back, of course I can do so.

Or maybe I'm missing something about playing this game?